Alvind0.HabitTracker/Database.cs

```diff
@@ -0,0 +1,676 @@
+            Console.WriteLine("Invalid date (format dd-MM-yy). Please try again.\n");
+            dateInput = Console.ReadLine();
+        }
+
+        return dateInput;
+    }
+
+    private static int GetNumber(string message)
+    {
+        Console.WriteLine(message);
+
+        string? numberInput = Console.ReadLine();
+        int output;
+        while (!int.TryParse(numberInput, out output) || Convert.ToInt32(numberInput) < 0)
+        {
+            Console.Clear();
+            Console.WriteLine("Invalid number (positive integers only). Please try again.\n");
+            numberInput = Console.ReadLine();
+        }
+        return output;
+    }
+    internal static void ViewStatistics(List<StatisticRecord> records, string habit)
+    {
+        Table table = new();
+        table.Border = TableBorder.Simple;
+        table.AddColumn("[bold]Habit[/]");
+        table.AddColumn("[bold]Total[/]");
+
+        foreach (StatisticRecord record in records)
+        {
+            table.AddRow(habit, $"{record.Total} {record.Unit}");
+        }
+        AnsiConsole.Write(table);
+    }
+
+    internal static void GetStatistics()
+    {
+        Console.Clear();
+        string habit;
+        GetHabits();
+        while (true)
+        {
+            habit = AnsiConsole.Prompt(
+                new TextPrompt<string>("Enter which habit to add a record to insert 0 to go back to Main Menu:\n"));
+            if (habit == "0")
+            {
+                Console.Clear();
+                return;
+            }
+            else if (!CheckIfHabitExists(habit))
+            {
+                Console.Clear();
+                Console.WriteLine("Habit does not exist.");
+                GetHabits();
+                continue;
+            }
+            break;
+        }
+
+        List<StatisticRecord> statistics = new();
+        string reportRange = Menu.StatisticsMenu();
+        string thisMonth = DateTime.Today.ToString("MM");
+        string thisYear = DateTime.Today.ToString("yy");
+
+        using (SqliteConnection connection = new(ConnectionString))
+        {
+            using (SqliteCommand tableCmd = connection.CreateCommand())
+            {
+                connection.Open();
+
+                switch (reportRange)
+                {
+                    case "All Time":
+                        tableCmd.CommandText = @"
+SELECT SUM(Quantity) AS Sum 
+FROM Habits 
+WHERE Habits.Habit = @habit";
+                        break;
+                    case "This Year":
+                        tableCmd.CommandText = @"
+SELECT SUM(Quantity) AS Sum 
+FROM Habits 
+WHERE Habit = @habit AND 
+Date LIKE '%' || @year";
+                        tableCmd.Parameters.AddWithValue("@year", thisYear);
+                        break;
+                    case "This Month":
+                        tableCmd.CommandText = @"
+SELECT SUM(Quantity) AS Sum 
+FROM Habits 
+WHERE Habit = @habit AND 
+Date LIKE '%' || @month || '-%'";
```

Comment:
This implementation causes records for the specified month of each year to be counted, not only records for "this month." For clarification, I added a record with the date 12-12-2012, and it is included even if the current month is December 2024.